docs/topics/testing/tools.txt

```diff
@@ -112,15 +112,20 @@ Making requests
 
 Use the ``django.test.Client`` class to make requests.
 
-.. class:: Client(enforce_csrf_checks=False, raise_request_exception=True, json_encoder=DjangoJSONEncoder, **defaults)
+.. class:: Client(enforce_csrf_checks=False, raise_request_exception=True, json_encoder=DjangoJSONEncoder, *, headers=None, **defaults)
 
-    It requires no arguments at time of construction. However, you can use
-    keyword arguments to specify some default headers. For example, this will
-    send a ``User-Agent`` HTTP header in each request::
+    It requires no arguments at time of construction. However, you can specify
+    some default headers through the ``headers`` parameter, or using keyword
+    arguments through the ``META`` keys. For example, these two examples are
+    equivalent and will send a ``User-Agent`` HTTP header in each request::
+
+        >>> c = Client(headers={'User-Agent': 'Mozilla/5.0'})
+
+    and
 
         >>> c = Client(HTTP_USER_AGENT='Mozilla/5.0')
```

Comment:
Some tweaks...

1. The opening sentence was super awkward. Turned into a more standard sentence describing what "the thing" (`Client`) is, plus an intro to the arguments.
2. Changed from prompt examples to standard python - it would be usual to use the test client from a prompt
3. Updated from Mozilla 5.0 to modern cURL as our example User-Agent. Browser User-Agent strings are a complete mess, but at least cURL remains simple.
4. Changed the arbitrary keyword docs to not override an HTTP header. We don't want to recommend `HTTP_*` going forwards for headers, or confuse people with "there are two ways to do things", so I think it's simplest that we show an example that's not covered by `headers`. Readers can still see from the WSGI spec that `HTTP_*` sets headers.

```suggestion
    A testing HTTP client. Takes several arguments that can customize behavior.
    
    ``headers`` allows you to specify default headers that will be sent with
    every request. For example, to set a ``User-Agent`` header::

        client = Client(headers={'User-Agent': 'curl/7.79.1'})

    Arbitrary keyword arguments in ``**defaults`` set WSGI `environ
    variables`_. For example, to set the script name::

        client = Client(SCRIPT_NAME='/app/')

    .. _environ variables: https://peps.python.org/pep-3333/#environ-variables
```

What do you think? Perhaps we could still have a sentence pointing out that `HTTP_*` sets headers to avoid people reading tests getting lost?

Reply:
> Perhaps we could still have a sentence pointing out that HTTP_* sets headers to avoid people reading tests getting lost?

I think that's a good idea 👍  We should not completely omit the `HTTP_*` way, but mention it to know it exists, works, but is not the preferred method anymore.